### PR TITLE
Add a note about replayEndPhase expectations in TCK

### PR DESCRIPTION
**Pull request by xstefank**
Signed-off-by: xstefank <xstefank122@gmail.com>

### CODE REVIEW

File: tck/running_the_tck.asciidoc [hunk 1]
```diff
@@ -109,3 +109,10 @@ Debugging is dependent on the TCK implementor. The implementor configures Arquil
 to deploy the test deployments and run the LRA client side implementation there. In general we can say that
 the runtime needs to define java debug properties `-Xdebug -Xrunjdwp:transport=dt_socket,address=8787,server=y,suspend=y`.
 Then debugger may be connected to port 8787 and to track the test execution.
+
+=== Additional notes
+
+* `TckTestBase#replayEndPhase(String)` method, used for `@Status` and `@Forget` calls
```
Comment (rdebusscher) [inline tck/running_the_tck.asciidoc]: > used for `@Status` and `@Forget` call verifications, expects that recovery happens on all active LRAs in a reasonably time frame

-> adjusted a few words.

Aren't we using the (configurable) timeouts also for these calls? Then we should refer to the parameter to change the factor for the timeout here also.

File: tck/running_the_tck.asciidoc [hunk 2]
```diff
@@ -109,3 +109,11 @@ Debugging is dependent on the TCK implementor. The implementor configures Arquil
...
 the runtime needs to define java debug properties `-Xdebug -Xrunjdwp:transport=dt_socket,address=8787,server=y,suspend=y`.
 Then debugger may be connected to port 8787 and to track the test execution.
+
+=== Additional notes
+
+* `TckTestBase#replayEndPhase(String)` method, used for `@Status` and `@Forget` call 
+verifications, expects that recovery happens on all active LRAs in a reasonable 
+time frame defined by the config property `lra.tck.recovery.timeout`. The 
+implementation must ensure that recovery is triggered for every active LRA at 
+approximately same time.
```
Comment (mmusgrov) [inline tck/running_the_tck.asciidoc]: The bit of text about "approximately same time" is just noise but I will approve anyway.

File: tck/running_the_tck.asciidoc [hunk 3]
```diff
@@ -109,3 +109,11 @@ Debugging is dependent on the TCK implementor. The implementor configures Arquil
 to deploy the test deployments and run the LRA client side implementation there. In general we can say that
 the runtime needs to define java debug properties `-Xdebug -Xrunjdwp:transport=dt_socket,address=8787,server=y,suspend=y`.
 Then debugger may be connected to port 8787 and to track the test execution.
+
+=== Additional notes
+
+* `TckTestBase#replayEndPhase(String)` method, used for `@Status` and `@Forget` call
```
Comment (mmusgrov) [inline tck/running_the_tck.asciidoc]: It is also used for testing participants that compensate in the background.

File: tck/running_the_tck.asciidoc (hunk 2 above)
Comment (xstefank) [reply tck/running_the_tck.asciidoc]: I am struggling how to state this. This is exactly the point of this PR. "lra.tck.recovery.timeout" sets the maximum time until which recovery should happen, let's say it is 2 minutes. But when recovery happens on ParticipatingTckResource then we have one second interval after which we check that actual TCK resource under test recovered also. So it's not possible to recover one LRA  after 10s and another after 110s. And this what I am trying to say with approximately same time. Any suggestions?

File: tck/running_the_tck.asciidoc (hunk 3 above)
Comment (xstefank) [reply tck/running_the_tck.asciidoc]: Thanks! I will add "for instance" in this sentence.

File: tck/running_the_tck.asciidoc (hunk 2 above)
Comment (mmusgrov) [reply tck/running_the_tck.asciidoc]: The previous paragraph is sufficient so just delete the second one.

File: tck/running_the_tck.asciidoc (hunk 1 above)
Comment (mmusgrov) [reply tck/running_the_tck.asciidoc]: Is this discussion resolved?

Comment (mmusgrov) [reply tck/running_the_tck.asciidoc]: @rdebusscher the updated text says:

> ...a reasonable time frame defined by the configuration property `lra.tck.recovery.timeout`.

Comment (mmusgrov) [reply tck/running_the_tck.asciidoc]: ie I think this PR is in a state where it can be committed - please could you add your approval?

**Pull request merged**